src/devices/DCMotor/DCMotor.cs

```diff
@@ -41,58 +41,130 @@ public virtual void Dispose()
+        /// </summary>
+        /// <param name="speedControlPin">Pin used to control the speed of the motor with software PWM</param>
+        /// <param name="controller">GPIO controller related to the pin</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(int speedControlPin, GpioController controller = null)
+        {
+            if (speedControlPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(speedControlPin));
+
+            controller = controller ?? new GpioController();
+            return new DCMotor2PinNoEnable(
+                new SoftwarePwmChannel(speedControlPin, 50, 0.0, controller: controller),
+                -1,
+                controller);
+        }
+
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in both directions.
+        /// </summary>
+        /// <param name="speedControlChannel">PWM channel used to control the speed of the motor</param>
+        /// <param name="directionPin">Pin used to control the direction of the motor</param>
+        /// <param name="controller">GPIO controller related to the pin</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(PwmChannel speedControlChannel, int directionPin, GpioController controller = null)
+        {
+            if (speedControlChannel == null)
+                throw new ArgumentNullException(nameof(speedControlChannel));
+
+            if (directionPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(directionPin));
+
+            return new DCMotor2PinNoEnable(speedControlChannel, directionPin, controller);
+        }
+        
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in both directions.
+        /// </summary>
+        /// <param name="speedControlPin">Pin used to control the speed of the motor with software PWM</param>
+        /// <param name="directionPin">Pin used to control the direction of the motor</param>
+        /// <param name="controller">GPIO controller related to the pins</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(int speedControlPin, int directionPin, GpioController controller = null)
+        {
+            if (speedControlPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(speedControlPin));
+
+            if (directionPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(directionPin));
+
+            controller = controller ?? new GpioController();
+            return new DCMotor2PinNoEnable(
+                new SoftwarePwmChannel(speedControlPin, 50, 0.0, controller: controller),
+                directionPin,
+                controller);
+        }
+
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in both directions.
```

Comment:
From the summary description, this factory method seems to be the exact same thing as the previous factory method. Why would I want to pass in otherDirectionPin? What do I gain with using 3 pin instead of 2 pin?

Reply:
Read README.md

Reply:
I did, it is not clear enough. It just say that we recommend 3-pin but doesn't state why. Also, we shouldn't expect consumers of this API to have to go read a README file in github. Instead they should be able to pick which API to call/use based on the xml docs that the API provides.